Document qlight CLI options (#212)
ConsenSys/doc.goquorum: docs/reference/cli-syntax.md

### `qlight.client.rpc.tls.key`

=== "Syntax"

    ```bash
    --qlight.client.rpc.tls.key <path>/<to>/<client_TLS_key_pem_file>
    ```

=== "Example"

    ```bash
    --qlight.client.rpc.tls.key client.TLS.key.pem
    ```

Path to the qlight client RPC client certificate private key.

### `qlight.client.serverNode`

=== "Syntax"

    ```bash
    --qlight.client.serverNode <nodeID>
    ```

=== "Example"

    ```bash
    --qlight.client.serverNode 0xc35c3...d615f
    ```

The node ID of the target server node.

### `qlight.client.serverNodeRPC`

=== "Syntax"

    ```bash
    --qlight.client.serverNodeRPC <URL>
    ```

=== "Example"

    ```bash
    --qlight.client.serverNodeRPC "http://127.0.0.1:8888"
    ```

The RPC URL of the target server node.

### `qlight.client.token.enabled`

=== "Syntax"

    ```bash
    --qlight.client.token.enabled
    ```

Enables the client to use a token when connecting to the qlight server.

### `qlight.client.token.management`

=== "Syntax"

    ```bash
    --qlight.client.token.management <string>
    ```

=== "Example"

    ```bash
    --qlight.client.token.management "none"
    ```

Mechanism used to refresh the token. Possible values:

> * `none` - Developer mode
> * `external` - A new token must be injected via the qlight RPC API
> * `client-security-plugin` - The client security plugin must be deployed/configured

Reviewer: ```suggestion
* `none` - Developer mode. The token is not refreshed.
* `external` - You must update the refreshed token in the running qlight client process by invoking the `qlight.setCurrentToken` RPC API.
* `client-security-plugin` - You must deploy the client security plugin, which periodically refreshes the access token.
```

Is there any documentation for the client security plugin?

### `qlight.client.token.value`

=== "Syntax"

    ```bash
    --qlight.client.token.value <TOKEN>
    ```

=== "Example"

>     ```bash
>     --qlight.client.token.value AYjcyMzY3ZDhiNmJkNTY
>     ```

Reviewer 2: ```suggestion
    ```bash
    --qlight.client.token.value AYjcyMzY3ZDhiNmJkNTY
    ```

    ```bash
    --qlight.client.token.value "bearer AYjcyMzY3ZDhiNmJkNTY"
    ```
```

Author: Fixed

Token the qlight client uses to connect to a server node.

### `qlight.server`

=== "Syntax"

    ```bash
    --qlight.server
    ```

Enables the [qlight server](../configure-and-manage/configure/qlight-node.md) P2P protocol.

### `qlight.server.p2p.maxpeers`

=== "Syntax"

    ```bash
    --qlight.server.p2p.maxpeers <INTEGER>
    ```

=== "Example"

    ```bash
    --qlight.server.p2p.maxpeers 10
    ```

Maximum number of qlight peers. The default is 10.

### `qlight.server.p2p.netrestrict`

=== "Syntax"

    ```bash
    --qlight.server.p2p.netrestrict <NETWORK MASK>
    ```

=== "Example"

    ```bash
    --qlight.server.p2p.netrestrict "xyz"
    ```

Restricts network communication to the given IP networks (CIDR masks).

### `qlight.server.p2p.permissioning`

=== "Syntax"

    ```bash
    --qlight.server.p2p.permissioning
    ```

Enables the qlight peers to check against a permissioned list and a disallowed list.

### `qlight.server.p2p.permissioning.prefix`

=== "Syntax"

    ```bash

>     --qlight.server.p2p.permissioning.prefix <prefix-filename>

Reviewer: ```suggestion
    --qlight.server.p2p.permissioning.prefix <prefix>
```
If my understanding is correct.

Author: I think just doing `<prefix>` is a little confusing, and spelling it out shows the proper formatting a little better.  I clarified it as `<prefix-defaultfilename>` and clarified the formatting in the description.

    ```

=== "Example"

    ```bash
    --qlight.server.p2p.permissioning.prefix "permissioned-nodes.json"
    ```

> Prefix for the permissioned-nodes.json and disallowed-nodes.json files.

Reviewer: ```suggestion
Prefix for the `permissioned-nodes.json` and `disallowed-nodes.json` files.
```

>     --qlight.server.p2p.permissioning.prefix "permissioned-nodes.json"

Reviewer: Isn't `permissioned-nodes.json` a default filename, not a prefix example?

Author: Good catch, fixed

> Prefix for the `permissioned-nodes.json` and `disallowed-nodes.json` files.

Reviewer: Maybe specify here the filename with prefix format (prefix, hyphen, default file name). Also the example is confusing, maybe use a prefix in the example, then in the description show what the resulting filename would be.

Author: FIxed

Port the qlight network listens to. The default is 30305.

### `qlight.tls`

=== "Syntax"

    ```bash
    --qlight.tls
    ```

Enables the qlight client P2P protocol to use TLS.

### `qlight.tls.cacerts`

=== "Syntax"

    ```bash
    --qlight.tls.cacerts <path>/<to>/<qlight_tls_cacert_file>
    ```

=== "Example"

    ```bash
    --qlight.tls.cacerts certfile.pem
    ```

Path to the certificate authorities file to use for validating P2P connection.

### `qlight.tls.cert`

=== "Syntax"

    ```bash
    --qlight.tls.cert` <path>/<to>/<qlight_tls_cert_file>
    ```

=== "Example"

    ```bash
    --qlight.tls.cert certfile.pem
    ```

Path to the certificate file to use for the qlight P2P connection.

### `qlight.tls.ciphersuites`

=== "Syntax"

    ```bash
    --qlight.tls.ciphersuites <STRING>
    ```

=== "Example"

    ```bash
    --qlight.tls.ciphersuites "CIPHER_SUITE_1,CIPHER_SUITE_2"
    ```

Cipher suites to use for the qlight P2P connection.

### `qlight.tls.clientauth`

=== "Syntax"

    ```bash
    --qlight.tls.clientauth <INTEGER>
    ```

=== "Example"

    ```bash
    --qlight.tls.clientauth 0
    ```

Sets the method the client is authenticated. Possible values:

> * 0=`NoClientCert` This is the default.

Reviewer: ```suggestion
* 0=`NoClientCert` (default)
```
